src/mito2/src/request.rs

```diff
@@ -100,32 +100,67 @@ pub struct WriteRequest {
     pub rows: Rows,
     /// Map column name to column index in `rows`.
     name_to_index: HashMap<String, usize>,
+    /// Whether each column has null.
+    has_null: Vec<bool>,
 }
 
 impl WriteRequest {
     /// Returns a new request.
-    pub fn new(region_id: RegionId, op_type: OpType, rows: Rows) -> WriteRequest {
-        let name_to_index = rows
-            .schema
-            .iter()
-            .enumerate()
-            .map(|(index, column)| (column.column_name.clone(), index))
-            .collect();
-        WriteRequest {
+    pub fn new(region_id: RegionId, op_type: OpType, rows: Rows) -> Result<WriteRequest> {
+        let mut name_to_index = HashMap::with_capacity(rows.schema.len());
+        for (index, column) in rows.schema.iter().enumerate() {
+            ensure!(
+                name_to_index
+                    .insert(column.column_name.clone(), index)
+                    .is_none(),
+                InvalidRequestSnafu {
+                    region_id,
+                    reason: format!("duplicate column {}", column.column_name),
+                }
+            );
+        }
+
+        let has_null = vec![false; rows.schema.len()];
+        let mut request = WriteRequest {
             region_id,
             op_type,
             rows,
             name_to_index,
-        }
+            has_null,
+        };
+        request.init()?;
+
+        Ok(request)
     }
 
-    /// Validate the request.
-    pub(crate) fn validate(&self) -> Result<()> {
-        // - checks whether the request is too large.
-        // - checks whether each row in rows has the same schema.
-        // - checks whether each column match the schema in Rows.
-        // - checks rows don't have duplicate columns.
-        unimplemented!()
+    /// Initializes and validates the request.
+    ///
+    /// Ensures rows match the schema.
+    fn init(&mut self) -> Result<()> {
```

Comment:
"init" a request looks a bit strange. Can we do this in `new()`?

Reply:
Removed in https://github.com/GreptimeTeam/greptimedb/pull/2135/commits/1e9f5bfcaa9f5dfdc0440e541aa7704111e74d8c